deprecate container, change docs to image_uri (#49752)
ray-project/ray: doc/source/ray-core/handling-dependencies.rst

  - Example: ``{"stop-on-exit": "true", "t": "cuda,cublas,cudnn", "ftrace": ""}``

> - ``image_uri`` (dict): Require a given (Docker) image, and the worker process will run in a container with this image.

Reviewer: ```suggestion
- ``image_uri`` (dict): Require a given Docker image. The worker process runs in a container with this image.
```

  - Example: ``{"image_uri": "anyscale/ray:2.31.0-py39-cpu"}``

>   Note: ``image_uri`` is experimental now. If you have some requirements or run into any problems, raise issues in `github <https://github.com/ray-project/ray/issues>`__.

Reviewer: ```suggestion
  Note: ``image_uri`` is experimental. If you have some requirements or run into any problems, raise issues in `github <https://github.com/ray-project/ray/issues>`__.
```
